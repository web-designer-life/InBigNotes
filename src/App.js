import React from 'react';
import './App.css';
import Container from './Container';

class App extends React.Component {
  render() {
      return (
          <div>
              <Container />
          </div>
      );
  }
}

export default App;
