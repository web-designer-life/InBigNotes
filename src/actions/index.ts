export const FETCH_NOTES_PENDING = 'FETCH_NOTES_PENDING';
export const FETCH_NOTES_SUCCESS = 'FETCH_NOTES_SUCCESS';
export const FETCH_NOTES_FAIL = 'FETCH_NOTES_FAIL';

export const ADD_NOTES_PENDING = 'ADD_NOTES_PENDING';
export const ADD_NOTES_SUCCESS = 'ADD_NOTES_SUCCESS';
export const ADD_NOTES_FAIL = 'ADD_NOTES_FAIL';

export const DELETE_NOTES_PENDING = 'DELETE_NOTES_PENDING';
export const DELETE_NOTES_SUCCESS = 'DELETE_NOTES_SUCCESS';
export const DELETE_NOTES_FAIL = 'DELETE_NOTES_FAIL';

export default { 
    FETCH_NOTES_PENDING,
    FETCH_NOTES_SUCCESS,
    FETCH_NOTES_FAIL,
    ADD_NOTES_PENDING, 
    ADD_NOTES_SUCCESS, 
    ADD_NOTES_FAIL, 
    DELETE_NOTES_PENDING, 
    DELETE_NOTES_SUCCESS, 
    DELETE_NOTES_FAIL,
};
