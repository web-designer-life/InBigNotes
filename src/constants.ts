export enum ROUTES {
	HOME = '/',
    NOTE = '/note',
	NOTE_CREATE = '/note/create',
};

export enum BUTTON_TYPES {
    BUTTON = 'button',
    SUBMIT = 'submit',
    RESET = 'reset',
};

export enum BUTTON_COLORS {
    GREEN = 'green',
    RED = 'red',
};

export enum BUTTON_TEXTS {
    CREATE = 'Create',
    DELETE = 'Delete',
    BACK = 'Back',
    SAVE = 'Save',
    UPDATE = 'Update',
    CANCEL = 'Cancel',
    CONFIRM = 'Confirm',
};

export enum MODAL_TEXTS {
    DELETE = 'Do you want to delete selected notes?',
    BACK = 'You have changes in your note. Do you want to exit to the main menu without saving?',
    SAVE = 'Do you want to save your note?',
    UPDATE = 'Do you want to update your note?',
    CANCEL = 'You have changes in your note. Do you want to cancel the changes?',
};

export const SOURCE = {
    ROUTES: {
        HOME: '/',
        NOTE: '/note',
        NOTE_CREATE: '/note/create',
    },
    BUTTONS: {
        TYPES: {
            BUTTON: 'button',
            SUBMIT: 'submit',
            RESET: 'reset',
        },
        COLORS: {
            GREEN: 'green',
            RED: 'red',
        },
        TEXTS: {
            CREATE: 'Create',
            DELETE: 'Delete',
            BACK: 'Back',
            SAVE: 'Save',
            UPDATE: 'Update',
            CANCEL: 'Cancel',
            CONFIRM: 'Confirm',
        },
    },
    MODAL: {
        TEXTS: {
            DELETE: 'Do you want to delete selected notes?',
            BACK: 'You have changes in your note. Do you want to exit to the main menu without saving?',
            SAVE: 'Do you want to save your note?',
            UPDATE: 'Do you want to update your note?',
            CANCEL: 'You have changes in your note. Do you want to cancel the changes?',
        },
    },
}
