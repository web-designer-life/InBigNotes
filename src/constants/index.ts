import { COLORS } from './colors';
import { FORMS } from './forms';
import { ROUTES } from './routes';
import { TEXTS } from './texts';
import { BUTTON_TYPES, ACTIONS } from './types';

export {
    COLORS,
    FORMS,
    ROUTES,
    TEXTS,
    BUTTON_TYPES, 
    ACTIONS,
}