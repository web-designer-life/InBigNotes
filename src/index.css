body {
    font-family: 'Lora', 'Arial', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    height: 100%;
    margin: 0;
    padding: 0;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}

#root {
    display: flex;
    flex-direction: column;
    max-width: 1280px;
    height: 100%;
    margin: auto;
}
