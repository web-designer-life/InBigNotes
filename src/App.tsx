import * as React from 'react';
import { GlobalStyle } from './Components/Style/GlobalStyle';
import Container from './Container';
import './App.css';

class App extends React.Component {
    render() {
        return (
            <>
                <GlobalStyle />
                <Container />
            </>
        );
    }
};
  
export default App;
