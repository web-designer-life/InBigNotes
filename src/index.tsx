import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import history, { store } from './store/store';
import App from './containers/App';

ReactDOM.render(
    <React.StrictMode>
        <Provider store={store}>
            <App history={history} />
        </Provider>
    </React.StrictMode>,
    document.getElementById('root')
);
