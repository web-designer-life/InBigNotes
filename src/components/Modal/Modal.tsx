import * as React from 'react';
import {
    ModalBackground,
    ModalContainer,
    ModalTitle,
    ButtonsContainer, 
} from './style';
import Button from '../Button/Button';
import { BUTTON_TYPES, BUTTON_COLORS } from '../../constants';

interface Props {
    modalText: string,
    buttonConfirmText: string, 
    buttonCancelText: string,
    active: boolean,
    onClose: () => void,
    action: () => void,
};

export default class Modal extends React.Component<Props> {
	render() {
        const {
            modalText,
            buttonConfirmText,
            buttonCancelText,
            active,
            onClose,
            action,
        } = this.props;

		return (
            active ?
            <ModalBackground>
                <ModalContainer>
                    <ModalTitle>{modalText}</ModalTitle>
                    <ButtonsContainer>
                        <Button 
                            type={BUTTON_TYPES.Button}
                            onClick={() => {
                                action();
                                onClose();
                            }}
                            text={buttonConfirmText} 
                            color={BUTTON_COLORS.Green}
                        />
                        <Button 
                            type={BUTTON_TYPES.Button}
                            onClick={onClose} 
                            text={buttonCancelText} 
                            color={BUTTON_COLORS.Red}
                        />
                    </ButtonsContainer>
                </ModalContainer>
            </ModalBackground> :
            null
		);
	}
};
