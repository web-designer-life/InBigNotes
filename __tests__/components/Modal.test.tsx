import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../src/components/Modal';
import { TEXTS } from '../../src/constants';

const activeModalProps = {
    modalText: TEXTS.MODAL.DELETE,
    buttonConfirmText: TEXTS.BUTTON.DELETE,
    buttonCancelText: TEXTS.BUTTON.CANCEL,
    active: true,
    onClose: () => {},
    action: () => {},
};

const inactiveModalProps = {
    modalText: TEXTS.MODAL.DELETE,
    buttonConfirmText: TEXTS.BUTTON.DELETE,
    buttonCancelText: TEXTS.BUTTON.CANCEL,
    active: false,
    onClose: () => {},
    action: () => {},
};

describe('Modal component', () => {
    it('should render active Modal component', () => {
        const component = shallow(<Modal {...activeModalProps} />);
        expect(component).toMatchSnapshot();
    });

    it('should render inactive Modal component', () => {
        const component = shallow(<Modal {...inactiveModalProps} />);
        expect(component).toMatchSnapshot();
    });
});
