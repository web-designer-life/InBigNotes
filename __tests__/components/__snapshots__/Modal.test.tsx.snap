// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component handlers handleClick should call method 1`] = `
<Modal
  action={[Function]}
  active={true}
  buttonCancelText="Cancel"
  buttonConfirmText="Delete"
  modalText="Do you want to delete selected notes?"
  onClose={[Function]}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "hhHtSZ",
            "rules": Array [
              "
	position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, .7);
    z-index: 999;
",
            ],
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bwzfXH hhHtSZ"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "PqmDJ",
                  "rules": Array [
                    "
	display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 20px;
    width: 400px;
    padding: 25px;
    margin: 25px;
    background-color: ",
                    "#FFFFFF",
                    ";
",
                  ],
                },
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat PqmDJ"
            >
              <styled.h2>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "cKwxgr",
                        "rules": Array [
                          "
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    line-height: 31px;
    text-align: center;
    margin: 0 0 25px 0;
    color: ",
                          "#5A5A5A",
                          ";
",
                        ],
                      },
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "h2",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <h2
                    className="sc-bxivhb cKwxgr"
                  >
                    Do you want to delete selected notes?
                  </h2>
                </StyledComponent>
              </styled.h2>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "kMSJaD",
                        "rules": Array [
                          "
	display: flex;
	justify-content: space-between;
    width: 100%;
    gap: 20px;
    @media (max-width: 989px) {
        flex-direction: column;
    }
",
                        ],
                      },
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-ifAKCX kMSJaD"
                  >
                    <Button
                      color="green"
                      disabled={false}
                      onClick={[Function]}
                      text="Delete"
                      type="button"
                    >
                      <styled.button
                        color="green"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <StyledComponent
                          color="green"
                          disabled={false}
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": false,
                                "lastClassName": "kHeIQK",
                                "rules": Array [
                                  "
    min-width: 170px;
    border-radius: 15px;
    color: ",
                                  "#FFFFFF",
                                  ";
    padding: 10px 25px;
    transition-duration: .1s;
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    &:hover {
        background-color: ",
                                  "#FFFFFF",
                                  ";
    }
    &:disabled {
        opacity: .5;
        border: 4px solid ",
                                  "#808080",
                                  ";
        background-color: ",
                                  "#808080",
                                  ";
        &:hover { 
            background-color: ",
                                  "#808080",
                                  ";
            color: ",
                                  "#FFFFFF",
                                  ";
            cursor: default;
        }
    }
    ",
                                  [Function],
                                  "
    ",
                                  [Function],
                                  "
    @media (max-width: 450px) {
        min-width: auto;
        width: 100%;
    }
",
                                ],
                              },
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          onClick={[Function]}
                          type="button"
                        >
                          <button
                            className="sc-bdVaJa mScxf"
                            color="green"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            Delete
                          </button>
                        </StyledComponent>
                      </styled.button>
                    </Button>
                    <Button
                      color="red"
                      disabled={false}
                      onClick={[Function]}
                      text="Cancel"
                      type="button"
                    >
                      <styled.button
                        color="red"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <StyledComponent
                          color="red"
                          disabled={false}
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": false,
                                "lastClassName": "kHeIQK",
                                "rules": Array [
                                  "
    min-width: 170px;
    border-radius: 15px;
    color: ",
                                  "#FFFFFF",
                                  ";
    padding: 10px 25px;
    transition-duration: .1s;
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    &:hover {
        background-color: ",
                                  "#FFFFFF",
                                  ";
    }
    &:disabled {
        opacity: .5;
        border: 4px solid ",
                                  "#808080",
                                  ";
        background-color: ",
                                  "#808080",
                                  ";
        &:hover { 
            background-color: ",
                                  "#808080",
                                  ";
            color: ",
                                  "#FFFFFF",
                                  ";
            cursor: default;
        }
    }
    ",
                                  [Function],
                                  "
    ",
                                  [Function],
                                  "
    @media (max-width: 450px) {
        min-width: auto;
        width: 100%;
    }
",
                                ],
                              },
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          onClick={[Function]}
                          type="button"
                        >
                          <button
                            className="sc-bdVaJa kHeIQK"
                            color="red"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            Cancel
                          </button>
                        </StyledComponent>
                      </styled.button>
                    </Button>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Modal>
`;

exports[`Modal component should render active Modal component 1`] = `
<styled.div>
  <styled.div>
    <styled.h2>
      Do you want to delete selected notes?
    </styled.h2>
    <styled.div>
      <Button
        color="green"
        disabled={false}
        onClick={[Function]}
        text="Delete"
        type="button"
      />
      <Button
        color="red"
        disabled={false}
        onClick={[Function]}
        text="Cancel"
        type="button"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Modal component should render inactive Modal component 1`] = `""`;
